tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/TestingFrameworkRetriesTests.cs

```diff
@@ -0,0 +1,134 @@
+
+    protected abstract string TrueAtThirdRetry { get; }
+
+    public virtual async Task FlakyRetries(string packageVersion)
+    {
+        EnvironmentHelper.EnableDefaultTransport();
+        var tests = new List<MockCIVisibilityTest>();
+        var testSuites = new List<MockCIVisibilityTestSuite>();
+        var testModules = new List<MockCIVisibilityTestModule>();
+
+        InjectSession(
+            out var sessionId,
+            out var sessionCommand,
+            out var sessionWorkingDirectory,
+            out var gitRepositoryUrl,
+            out var gitBranch,
+            out var gitCommitSha);
+
+        try
+        {
+            var retryCount = 5;
+            SetEnvironmentVariable(ConfigurationKeys.CIVisibility.FlakyRetryEnabled, "1");
+            SetEnvironmentVariable(ConfigurationKeys.CIVisibility.FlakyRetryCount, retryCount.ToString());
+
+            using var agent = EnvironmentHelper.GetMockAgent();
+            agent.EventPlatformProxyPayloadReceived += (sender, e) =>
+            {
+                if (e.Value.PathAndQuery.EndsWith("api/v2/libraries/tests/services/setting"))
+                {
+                    e.Value.Response = new MockTracerResponse(SettingsJson, 200);
+                    return;
+                }
+
+                if (e.Value.PathAndQuery.EndsWith("api/v2/citestcycle"))
+                {
+                    var payload = JsonConvert.DeserializeObject<MockCIVisibilityProtocol>(e.Value.BodyInJson);
+                    if (payload.Events?.Length > 0)
+                    {
+                        foreach (var @event in payload.Events)
+                        {
+                            if (@event.Content.ToString() is { } eventContent)
+                            {
+                                if (@event.Type == SpanTypes.Test)
+                                {
+                                    tests.Add(JsonConvert.DeserializeObject<MockCIVisibilityTest>(eventContent));
+                                }
+                                else if (@event.Type == SpanTypes.TestSuite)
+                                {
+                                    testSuites.Add(JsonConvert.DeserializeObject<MockCIVisibilityTestSuite>(eventContent));
+                                }
+                                else if (@event.Type == SpanTypes.TestModule)
+                                {
+                                    testModules.Add(JsonConvert.DeserializeObject<MockCIVisibilityTestModule>(eventContent));
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            using var processResult = await RunDotnetTestSampleAndWaitForExit(
+                                          agent,
+                                          packageVersion: packageVersion);
+
+            // 1 Module
+            testModules.Should().HaveCount(1);
```

Comment:
No snapshots? 😉

Reply:
no, tags contents are already covered by normal evp and early flake detection tests... I don't see the value of having them here as well compared to the complexity they add. For retries I think the presence of the events is enough.